yorkie/backend/pubsub/mempubsub/mempubsub.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mempubsub
+
+import (
+	"sync"
+
+	"github.com/yorkie-team/yorkie/pkg/document/time"
+	"github.com/yorkie-team/yorkie/pkg/log"
+	"github.com/yorkie-team/yorkie/pkg/types"
+	"github.com/yorkie-team/yorkie/yorkie/backend/pubsub"
+)
+
+// Subscriptions is a collection of subscriptions that subscribe to a specific
+// topic.
+type Subscriptions struct {
+	internalMap map[string]*pubsub.Subscription
+}
+
+func newSubscriptions() *Subscriptions {
+	return &Subscriptions{
+		internalMap: make(map[string]*pubsub.Subscription),
+	}
+}
+
+// Add adds the given subscription.
+func (s *Subscriptions) Add(sub *pubsub.Subscription) {
+	s.internalMap[sub.ID()] = sub
+}
+
+// Map returns the internal map of this Subscriptions.
+func (s *Subscriptions) Map() map[string]*pubsub.Subscription {
+	return s.internalMap
+}
+
+// Delete deletes the subscription of the given id.
+func (s *Subscriptions) Delete(id string) {
+	if subscription, ok := s.internalMap[id]; ok {
+		subscription.Close()
+	}
+	delete(s.internalMap, id)
+}
+
+// MemPubSub is the memory implementation of MemPubSub, used for single agent or
+// tests.
+type MemPubSub struct {
+	mu               *sync.RWMutex
+	subscriptionsMap map[string]*Subscriptions
+}
+
+// New creates an instance of MemoryPubSub.
+func New() *MemPubSub {
+	return &MemPubSub{
+		mu:               &sync.RWMutex{},
+		subscriptionsMap: make(map[string]*Subscriptions),
```

Comment:
Both `subscriptions` and `Map` have multiple data representations. So `subscriptionMap` or `subscriptions` looks good, what do you think?

Reply:
Thanks for the comments. But `subscriptionsMap` represents a map of `subscriptions` not a map of `subscription`. 
So `subscriptionMap` or `subscriptions` is not appropriate.